File: HW_2/train.cpp
#include "train.h"

int RandInt(int min, int max);

// Write train information in file.
void train::Write(FILE *file) {
    fprintf(file, "%s", "train ");
    transport::Write(file);
    fprintf(file, "%c%d", ' ', this->carsNumber);
}

// Read train information from file.
void train::Read(FILE *file) {
    transport::Read(file);
    fscanf(file, "%d", &this->carsNumber);
}

// Generate random train.
void train::Generate() {
    transport::Generate();
    this->carsNumber = RandInt(1, 50);
}

File: HW_3/Train.py
from Transport import Transport
import random


class Train(Transport):
    def __init__(self, params=None):
        super().__init__(params)
        if params is None:
            self.cars_number = 0
        else:
            self.cars_number = int(params[2])

    def generate(self):
        super().generate()
        self.cars_number = random.randint(1, 100)
        return self

    def __str__(self):
        return f"Train {super().__str__()} {self.cars_number}"
